src/react/hooks/__tests__/useSubscription.test.tsx

```diff
@@ -1123,6 +1125,231 @@ followed by new in-flight setup", async () => {
 
     unmount();
   });
+
+  describe.only("errorPolicy", () => {
+    function setup() {
+      const subscription: TypedDocumentNode<{ totalLikes: number }, {}> = gql`
+        subscription ($id: ID!) {
+          totalLikes
+        }
+      `;
+      const errorBoundaryOnError = jest.fn();
+      const link = new MockSubscriptionLink();
+      const client = new ApolloClient({
+        link,
+        cache: new Cache(),
+      });
+      const ProfiledHook = profileHook(
+        (options: SubscriptionHookOptions<{ totalLikes: number }, {}>) =>
+          useSubscription(subscription, options)
+      );
+      const wrapper = ({ children }: { children: any }) => (
+        <ApolloProvider client={client}>
+          <ErrorBoundary onError={errorBoundaryOnError} fallback={<>error</>}>
+            {children}
+          </ErrorBoundary>
+        </ApolloProvider>
+      );
+      const graphQlErrorResult: MockedSubscriptionResult = {
+        result: {
+          data: { totalLikes: 42 },
+          errors: [new GraphQLError("test")],
+        },
+      };
+      const protocolErrorResult: MockedSubscriptionResult = {
+        error: new ApolloError({ errorMessage: "test" }),
+      };
+      return {
+        client,
+        link,
+        errorBoundaryOnError,
+        ProfiledHook,
+        wrapper,
+        graphQlErrorResult,
+        protocolErrorResult,
+      };
+    }
+    describe("GraphQL error", () => {
+      it.each([undefined, "none"] as const)(
+        "`errorPolicy: '%s'`: returns `{ error }`, calls `onError`",
+        async (errorPolicy) => {
+          const {
+            ProfiledHook,
+            wrapper,
+            link,
+            graphQlErrorResult,
+            errorBoundaryOnError,
+          } = setup();
+          const onData = jest.fn();
+          const onError = jest.fn();
+          render(
+            <ProfiledHook
+              errorPolicy={errorPolicy}
+              onError={onError}
+              onData={onData}
+            />,
+            {
+              wrapper,
+            }
+          );
+
+          await ProfiledHook.takeSnapshot();
+          link.simulateResult(graphQlErrorResult);
+          {
+            const snapshot = await ProfiledHook.takeSnapshot();
+            expect(snapshot).toStrictEqual({
+              loading: false,
+              error: new ApolloError({ errorMessage: "test" }),
```

Comment:
```suggestion
              error: new ApolloError({ 
                errorMessage: "test",
                graphQLErrors: [new GraphQLError("test")],
              }),
```

Could we also ensure this error has `graphQLErrors` defined correctly?

Reply:
Phew... not with how Jest natively does it... but working on it.

For comparison:
https://github.com/jestjs/jest/blob/c04d13d7abd22e47b0997f6027886aed225c9ce4/packages/expect-utils/src/jasmineUtils.ts#L91-L94